cardinal/server/server.go

```diff
@@ -355,6 +358,78 @@ func registerReadHandlerSwagger(world *ecs.World, api *untyped.API, handler *Han
 		getListTxReceiptsReplyFromRequest(world),
 	)
 
+	cqlHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		mapStruct, ok := params.(map[string]interface{})
+		if !ok {
+			return nil, errors.New("invalid parameter input, map could not be created")
+		}
+		cqlRequestUntyped, ok := mapStruct["cql"]
+		if !ok {
+			return nil, errors.New("cql body parameter could not be found")
+		}
+		cqlRequest, ok := cqlRequestUntyped.(map[string]interface{})
+		if !ok {
+			return middleware.Error(422, fmt.Errorf("json is invalid")), nil
+		}
+		cqlStringUntyped, ok := cqlRequest["CQL"]
+		if !ok {
+			return middleware.Error(422, fmt.Errorf("json is invalid")), nil
+		}
+		cqlString, ok := cqlStringUntyped.(string)
+		if !ok {
+			return middleware.Error(422, fmt.Errorf("json is invalid")), nil
+		}
+		resultFilter, err := cql.CQLParse(cqlString, world.GetComponentByName)
+		if err != nil {
+			return middleware.Error(422, err), nil
+		}
+
+		result := make([]cql.QueryResponse, 0)
+
+		ecs.NewQuery(resultFilter).Each(world, func(id storage.EntityID) bool {
+			var entity storage.Entity
+			entity, err = world.Entity(id)
+			if err != nil {
+				return false
+			}
+			components := entity.GetComponents(world)
+			resultElement := cql.QueryResponse{
+				id,
+				make([]json.RawMessage, 0),
+			}
+
+			//The way our framework is set up it's not designed to retrieve components dynamically at runtime.
+			//As a result we have to use reflection which is generally bad and expensive.
```

Comment:
i wonder if we can just make a new method on the interface IComponentType that returns `interface{} / any` for component data, which is JSON marshal-able. 

would require passing in `world` which in turn would cause circular dep, but maybe we don't need `IComponentType` in its own package? maybe we can make a new ticket and follow up afterwards if that adds too much overhead to this PR

Reply:
Yeah this makes sense. I see what you're saying. I also agree I'll put this in another ticket.